### DISCUSSION TITLE
Change default root password

### DISCUSSION THREAD

**Discussion by heqins** [milvus-io/milvus#22736] [Q&A and General discussion] [answered]
On this page(https://milvus.io/docs/authenticate.md), it says we should change the password of the root user when we start Milvus for the first time. But with docker(https://github.com/milvus-io/milvus/releases/download/v2.2.3/milvus-standalone-docker-compose.yml), I didn't find any place to change the password.

**Comment by yhmo** [comment, marked as answer]
The https://milvus.io/docs/authenticate.md is introducing how to do RBAC(Role-Based Access Control). After RBAC is enabled, user can use username/password to access specific Milvus resources(collection, partition) by SDK.
The root user is predefined with name=root and password=Milvus, you can use RBAC interfaces to change its password:
```
from pymilvus import utility
from pymilvus import connections
connections.connect(
    alias='default',
    host='localhost',
    port='19530',
    user='root',
    password='Milvus',
)
utility.reset_password('root', 'Milvus', 'new_password', using='default') 
```

**Reply by heqins** [reply]
Thanks

**Comment by TaoWolf** [comment]
Title: "Unable to Add common.security.authorizationEnabled in Docker Configuration File"

Issue Description:
"In the Docker configuration file for Milvus, I am unable to add the common.security.authorizationEnabled option. Regardless of where I place this option in the configuration file, it results in an error. This makes it difficult to enable authorization features when deploying Milvus via Docker."

**Reply by yhmo** [reply]
You cannot add the "common.security.authorizationEnabled" into docker-compose.yml.
This doc tells how to configure milvus standalone: https://milvus.io/docs/configure-docker.md

In milvus container, there is a file milvus.yaml under the /milvus/configs/milvus.yaml, all the runtime configurations are defined in this file. And all these configurations have default values.

To change milvus configuration, we need to change the values of this file.
Step 1: mapping the internal milvus.yaml to anther milvus.yaml outside of the container
Step 2: modify the outside file
Step 3: start the server

**Reply by TaoWolf** [reply]
Thanks，but……      
However, if I set it up this way, I am unable to log in using any username and password. What should I do?

**Reply by TaoWolf** [reply]
![image](https://github.com/milvus-io/milvus/assets/16403888/45048f47-f8f1-4f3f-9c8a-b641b00a9100)

**Reply by TaoWolf** [reply]
ERROR: CheckHealth RPC failed!
in JAVA:
private static final MilvusServiceClient milvusClient = new MilvusServiceClient(
  ConnectParam.newBuilder()
    .withHost("xx.xxx.xxx.xx")
    .withPort(19530)
    .withDatabaseName(dbName)
    .withAuthorization("root","Milvus")
    .build()
);

**Reply by TaoWolf** [reply]
Thanks，OK！
